editor/store/persist.js

```diff
@@ -34,17 +34,17 @@ export function withRehydratation( reducer, reducerKey ) {
  *
  * This should be executed after the reducer's registration.
  *
- * @param {Object} store      Store to enhance.
- * @param {string} reducerKey The reducer key to persist (example: reducerKey.subReducerKey).
- * @param {string} storageKey The storage key to use.
- * @param {Object} defaults   Default values of the reducer key.
+ * @param {Object}   store      Store to enhance.
+ * @param {Function} reducer    The reducer function. Used to get default values and to allow custom serialization by the reducers.
+ * @param {string}   reducerKey The reducer key to persist (example: reducerKey.subReducerKey).
+ * @param {string}   storageKey The storage key to use.
  */
-export function loadAndPersist( store, reducerKey, storageKey, defaults = {} ) {
+export function loadAndPersist( store, reducer, reducerKey, storageKey ) {
 	// Load initially persisted value
 	const persistedString = window.localStorage.getItem( storageKey );
 	if ( persistedString ) {
 		const persistedState = {
-			...defaults,
+			...get( reducer( undefined, { type: 'DEFAULTS' } ), reducerKey ),
```

Comment:
Not sure it's a convention we care to adopt, but Redux goes out of its way to try to make the `type` distinct so as not to conflict with what could exist as an application-handled "DEFAULTS" type here. The Redux initialization type is `@@INIT`.